storage/storage.cpp

```diff
@@ -614,14 +614,28 @@ void Storage::DownloadNextFile(QueuedCountry const & country)
 {
   TCountryId const & countryId = country.GetCountryId();
   CountryFile const & countryFile = GetCountryFile(countryId);
+  auto const opt = country.GetCurrentFileOptions();
 
-  string const filePath = GetFileDownloadPath(countryId, country.GetCurrentFileOptions());
+  string const readyFilePath = GetFileDownloadPath(countryId, opt);
   uint64_t size;
+  auto & p = GetPlatform();
+
+  // Since a downloaded valid diff file may be either with .diff or .diff.ready extension,
+  // we have to check these both cases in order to find
+  // the diff file which is ready to apply.
+  // If there is a such file we have to cause the success download scenario.
+  bool isDownloadedDiff = false;
+  if (opt == MapOptions::Diff)
+  {
+    string filePath = readyFilePath;
+    my::GetNameWithoutExt(filePath);
+    isDownloadedDiff = p.GetFileSizeByFullPath(filePath, size);
+  }
 
   // It may happen that the file already was downloaded, so there're
   // no need to request servers list and download file.  Let's
   // switch to next file.
-  if (GetPlatform().GetFileSizeByFullPath(filePath, size))
+  if (p.GetFileSizeByFullPath(readyFilePath, size) || isDownloadedDiff)
```

Comment:
Please, do not implicitly convert integrals to bools. Why not to use function, like, IsFileExistsByFullPath, or something like that?

Reply:
OK, I get it. But would it be safer to swap operands here?

Reply:
Because GetFileSizeByFullPath returns bool and we also need this size value. 
One thing that I should do it's checking isDownloadedDiff at first.